bin/ch/WScriptJsrt.cpp

```diff
@@ -34,6 +34,25 @@
 #endif // FreeBSD or unix ?
 #endif // _WIN32 ?
 
+
+struct ThreadLocalData
+{
+    ThreadLocalData()
+        :threadData(nullptr)
+    {
+    }
+    ~ThreadLocalData()
+    {
+        if (threadData)
+        {
+            delete threadData;
```

Comment:
Please add an `xplat-todo: semaphore could be nice here` note to the places. I'm updating xplat locking backend hence I can visit those comments later